src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java

```diff
@@ -146,16 +173,15 @@ private BitbucketWebhook process(WebhookRegisterRequest request,
                 .stream()
                 .filter(hook -> hook.getEvents().equals(Collections.singleton(REPO_REF_CHANGE.getEventId())))
                 .collect(toList());
+
         List<BitbucketWebhook> webhookWithPROnly = ownedHooks
                 .stream()
-                .filter(hook -> hook.getEvents().equals(Collections.singleton(PULL_REQUEST_OPENED_EVENT.getEventId())))
+                .filter(hook -> hook.getEvents().equals(AllPREventIds))
                 .collect(toList());
 
-        Set<String> refAndPullEvent = new HashSet<>(Arrays.asList(REPO_REF_CHANGE.getEventId(), PULL_REQUEST_OPENED_EVENT.getEventId()));
-
         List<BitbucketWebhook> webhookWithRepoRefChangeOrPR = ownedHooks
                 .stream()
-                .filter(hook -> hook.getEvents().stream().anyMatch(refAndPullEvent::contains))
+                .filter(hook -> hook.getEvents().stream().anyMatch(refAndPREventIds::contains))
                 .collect(toList());
 
         if (ownedHooks.isEmpty() ||
```

Comment:
I'm finding this conditional a bit hard to read, as it requires referring back through the method a lot. One suggestion might be to do it like this:
`(ownedHooks.stream().noneMatch(hook -> hook.getEvents().contains(MIRROR_SYNCHRONIZED_EVENT)) && events.contains(MIRROR_SYNCHRONIZED_EVENT))`
You can do this for each conditional you have here, which removes the need for the three or four lists you have above-- but that comes at the cost of making the conditional more complex (it _definitely_ needs some comments).
Readabiliity suggestion though, if you prefer the declared lists stick with them.